File: ch.hsr.ifs.test.framework/src/ch/hsr/ifs/test/framework/preference/PreferenceInitializer.java
/*******************************************************************************
 * Copyright (c) 2007-2011, IFS Institute for Software, HSR Rapperswil,
 * Switzerland, http://ifs.hsr.ch
 * 
 * Permission to use, copy, and/or distribute this software for any
 * purpose without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 ******************************************************************************/
package ch.hsr.ifs.test.framework.preference;

import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer;
import org.eclipse.jface.preference.IPreferenceStore;

import ch.hsr.ifs.test.framework.TestFrameworkPlugin;

/**
 * @author Emanuel Graf
 *
 */
public class PreferenceInitializer extends AbstractPreferenceInitializer {

	@Override
	public void initializeDefaultPreferences() {
		IPreferenceStore store = TestFrameworkPlugin.getDefault().getPreferenceStore();
		store.setDefault(PreferenceConstants.SHOW_WHITESPACES, false);
	}

}

File: CUTEProject/bundles/ch.hsr.ifs.testframework/src/ch/hsr/ifs/testframework/preference/PreferenceInitializer.kt
/*******************************************************************************
 * Copyright (c) 2007-2011, IFS Institute for Software, HSR Rapperswil,
 * Switzerland, http://ifs.hsr.ch
 *
 * Permission to use, copy, and/or distribute this software for any
 * purpose without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 ******************************************************************************/
package ch.hsr.ifs.testframework.preference;

import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer;
import org.eclipse.jface.preference.IPreferenceStore;

import ch.hsr.ifs.testframework.TestFrameworkPlugin;


/**
 * @author Emanuel Graf
 *
 */
class PreferenceInitializer : AbstractPreferenceInitializer() {

   override fun initializeDefaultPreferences() =  TestFrameworkPlugin.default?.getPreferenceStore()?.setDefault(SHOW_WHITESPACES, false) ?: Unit

}
